presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsRecordSet.java

```diff
@@ -0,0 +1,188 @@
+
+    public TpcdsRecordSet(Results results, List<Column> columns)
+    {
+        requireNonNull(results, "results is null");
+
+        this.results = results;
+        this.columns = ImmutableList.copyOf(columns);
+        this.columnTypes = ImmutableList.copyOf(this.columns.stream().map(column -> getPrestoType(column.getType())).collect(Collectors.toList()));
+    }
+
+    @Override
+    public List<Type> getColumnTypes()
+    {
+        return columnTypes;
+    }
+
+    @Override
+    public RecordCursor cursor()
+    {
+        return new TpcdsRecordCursor(results.iterator(), columns);
+    }
+
+    public class TpcdsRecordCursor
+            implements RecordCursor
+    {
+        private final Iterator<List<List<String>>> rows;
+        private final List<Column> columns;
+        private List<String> row;
+        private boolean closed;
+
+        public TpcdsRecordCursor(Iterator<List<List<String>>> rows, List<Column> columns)
+        {
+            this.rows = rows;
+            this.columns = columns;
+        }
+
+        @Override
+        public long getTotalBytes()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getCompletedBytes()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getReadTimeNanos()
+        {
+            return 0;
+        }
+
+        @Override
+        public Type getType(int field)
+        {
+            return getPrestoType(columns.get(field).getType());
+        }
+
+        @Override
+        public boolean advanceNextPosition()
+        {
+            if (closed || !rows.hasNext()) {
+                closed = true;
+                row = null;
+                return false;
+            }
+
+            row = rows.next().get(0);
+            return true;
+        }
+
+        @Override
+        public boolean getBoolean(int field)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long getLong(int field)
+        {
+            checkState(row != null, "No current row");
+            Column column = columns.get(field);
+            if (column.getType().getBase() == ColumnType.Base.DATE) {
+                return LocalDate.parse(row.get(column.getPosition())).toEpochDay();
+            }
+            if (column.getType().getBase() == ColumnType.Base.TIME) {
+                return LocalTime.parse(row.get(column.getPosition())).toNanoOfDay() / 1000000;
+            }
+            if (column.getType().getBase() == ColumnType.Base.INTEGER) {
+                return parseInt(row.get(column.getPosition()));
+            }
+            if (column.getType().getBase() == ColumnType.Base.DECIMAL) {
+                DecimalParseResult decimalParseResult = Decimals.parse(row.get(column.getPosition()));
+                return rescale((Long) decimalParseResult.getObject(), decimalParseResult.getType().getScale(), ((DecimalType) columnTypes.get(field)).getScale());
```

Comment:
Why do we need to rescale? Wouldn't `Decimals.parse()` return the correct scale?